File: core/repository/repository_reconstruction_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.parsers import parse_source

from .reconstruction import (
    build_dependency_lineage,
    build_deployment_graph,
    build_event_graph,
    build_runtime_graph,
    classify_architecture,
    infer_ownership_domains,
    reconstruct_monorepo,
    reconstruct_topology,
)


def reconstruct_repository(text: str, source_url: str = "", paths: list | None = None) -> Dict[str, Any]:
    parsed = parse_source(text, path=source_url or "repository")
    path_list = paths if isinstance(paths, list) else []
    topology = reconstruct_topology(path_list)
    events = build_event_graph(text)
    runtime = build_runtime_graph(
        parsed.get("runtime", {}).get("runtimes", []),
        topology.get("services", []),
    )
    deployment = build_deployment_graph(text)
    return {
        "parser": parsed,
        "topology": topology,
        "events": events,
        "runtime_graph": runtime,
        "deployment": deployment,
        "architecture": classify_architecture(topology, events, deployment),
        "monorepo": reconstruct_monorepo(path_list),
        "ownership": infer_ownership_domains(path_list),
        "dependency_lineage": build_dependency_lineage(
            parsed.get("dependencies", {}).get("dependencies", [])
        ),
    }

File: src/repository/repositoryReconstructionEngine.ts
/**
 * Converted from Python: core/repository/repository_reconstruction_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { parseSource } from "../parsers/index.js";
import { buildDependencyLineage, buildDeploymentGraph, buildEventGraph, buildRuntimeGraph, classifyArchitecture, inferOwnershipDomains, reconstructMonorepo, reconstructTopology } from "./reconstruction/index.js";

export function reconstructRepository(text: any, source_url: any = "", paths: any = null): any {
  var parsed: any = parseSource(text, py.or2(source_url, () => ("repository")));
  var path_list: any = ((Array.isArray(paths)) ? paths : []);
  var topology: any = reconstructTopology(path_list);
  var events: any = buildEventGraph(text);
  var runtime: any = buildRuntimeGraph(py.get(py.get(parsed, "runtime", {}), "runtimes", []), py.get(topology, "services", []));
  var deployment: any = buildDeploymentGraph(text);
  return {"parser": parsed, "topology": topology, "events": events, "runtime_graph": runtime, "deployment": deployment, "architecture": classifyArchitecture(topology, events, deployment), "monorepo": reconstructMonorepo(path_list), "ownership": inferOwnershipDomains(path_list), "dependency_lineage": buildDependencyLineage(py.get(py.get(parsed, "dependencies", {}), "dependencies", []))};
}
export { buildDependencyLineage, buildDeploymentGraph, buildEventGraph, buildRuntimeGraph, classifyArchitecture, inferOwnershipDomains, parseSource, reconstructMonorepo, reconstructTopology };
